NotesApi/V1/Domain/Queries/CreateNoteRequest.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace NotesApi.V1.Domain.Queries
+{
+    public class CreateNoteRequest
+    {
+        [Required]
+        [MaxLength(500)]
+        public string Description { get; set; }
+
+        public string TargetType { get; set; }
+
+        public string TargetId { get; set; }
```

Comment:
Maybe use the actual types here (enum and Guid). That was MVC effectively does any validation for you when the request comes in.